api/v1alpha1/usersignup_types.go

```diff
@@ -196,6 +196,51 @@ type UserSignupSpec struct {
 	// a new IdP provider client, and contains the user's "original-sub" claim
 	// +optional
 	OriginalSub string `json:"originalSub,omitempty"`
+
+	// IdentityClaims contains as-is claim values extracted from the user's access token
+	// +optional
+	IdentityClaims IdentityClaimsEmbedded `json:"identityClaims,omitempty"`
+}
+
+// IdentityClaimsEmbedded is used to define a set of SSO claim values that we are interested in storing
+// +k8s:openapi-gen=true
+type IdentityClaimsEmbedded struct {
+
+	// PropagatedClaims
+	PropagatedClaims PropagatedClaims `json:",inline"`
+
+	// Sub contains the value of the 'sub' claim
+	Sub string `json:"sub,omitempty"`
+
+	// Email contains the user's email address
+	Email string `json:"email,omitempty"`
```

Comment:
```suggestion
	// Email contains the user's email address
	// +optional
	Email string `json:"email,omitempty"`
```

Reply:
Email is not actually optional claim.

Reply:
good point. in that case:
```suggestion
	// Email contains the user's email address
	Email string `json:"email"`
```

Reply:
@alexeykazakov informed me today that email was not optional